src/NeoSharp.Application/Client/Prompt.cs

```diff
@@ -106,78 +89,37 @@ public partial class Prompt : IPrompt
         static Prompt()
         {
             _commandCache = new Dictionary<string[], PromptCommandAttribute>();
-            var commandAutocompleteCache = new Dictionary<string, List<ParameterInfo[]>>();
-
-            foreach (var mi in typeof(Prompt).GetMethods
-                (
-                BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Instance
-                ))
-            {
-                var atr = mi.GetCustomAttribute<PromptCommandAttribute>();
-                if (atr == null) continue;
-
-                atr.SetMethod(mi);
-
-                _commandCache.Add(atr.Commands, atr);
-
-                if (commandAutocompleteCache.ContainsKey(atr.Command))
-                {
-                    commandAutocompleteCache[atr.Command].Add(mi.GetParameters());
-                }
-                else
-                {
-                    var ls = new List<ParameterInfo[]>
-                    {
-                        mi.GetParameters()
-                    };
-                    commandAutocompleteCache.Add(atr.Command, ls);
-                }
-            }
-
-            _commandAutocompleteCache = new AutoCommandComplete(commandAutocompleteCache);
+            _commandCache.Cache(typeof(Prompt), out _commandAutocompleteCache);
         }
 
         #endregion
 
         /// <summary>
         /// Constructor
         /// </summary>
+        /// <param name="container">Container</param>
         /// <param name="consoleReaderInit">Console reader init</param>
         /// <param name="consoleWriterInit">Console writer init</param>
-        /// <param name="loggerFactory">Logget factory</param>
         /// <param name="logger">Logger</param>
-        /// <param name="networkManagerInit">Network manger init</param>
-        /// <param name="serverInit">Server</param>
-        /// <param name="rpcInit">Rpc server</param>
-        /// <param name="serializer">Binary serializer</param>
         /// <param name="blockchain">Blockchain</param>
-        /// <param name="blockPool">Block pool</param>
         /// <param name="walletManager"></param>
         /// <param name="vmFactory">VM Factory</param>
-        public Prompt(
+        public Prompt
+            (
+            IContainer container,
```

Comment:
this is not solution ... IContainer should never been injected

Reply:
I only be able to inject collection with generic types, what is the right way to register a collection of all types?

```csharp

 var promptHandlerTypes = typeof(IPromptController).Assembly
               .GetExportedTypes()
               .Where(t => t.IsClass && !t.IsAbstract && typeof(IPromptController).IsAssignableFrom(t))
               .ToArray();

            containerBuilder.RegisterSingleton(() => new PromptControllerFactory(promptHandlerTypes));

```